doctr/models/detection/differentiable_binarization/base.py

```diff
@@ -284,23 +288,24 @@ def build_target(
 
         input_dtype = next(iter(target[0].values())).dtype if len(target) > 0 else np.float32
 
+        h: int
+        w: int
         if channels_last:
-            h, w = output_shape[1:-1]
-            target_shape = (output_shape[0], output_shape[-1], h, w)  # (Batch_size, num_classes, h, w)
+            h, w, num_classes = output_shape
         else:
-            h, w = output_shape[-2:]
-            target_shape = output_shape  # (Batch_size, num_classes, h, w)
+            num_classes, h, w = output_shape
+        target_shape = (len(target), num_classes, h, w)
+
         seg_target: np.ndarray = np.zeros(target_shape, dtype=np.uint8)
         seg_mask: np.ndarray = np.ones(target_shape, dtype=bool)
         thresh_target: np.ndarray = np.zeros(target_shape, dtype=np.float32)
-        thresh_mask: np.ndarray = np.ones(target_shape, dtype=np.uint8)
+        thresh_mask: np.ndarray = np.zeros(target_shape, dtype=np.uint8)
```

Comment:
@odulcy-mindee @charlesmindee This was not correct before now the mask is generated correctly before it was ones and we also filled it with ones :sweat_smile:

Reply:
ah, oups :sweat_smile: